include/session/config/conversations.h

```diff
@@ -0,0 +1,190 @@
+#pragma once
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "base.h"
+#include "profile_pic.h"
+
+
+enum CONVO_EXPIRATION_MODE {
+    EXPIRATION_NONE = 0,
+    EXPIRATION_AFTER_SEND = 1,
+    EXPIRATION_AFTER_READ = 2,
+};
+
+
+typedef struct convo_one_to_one {
+    char session_id[67];  // in hex; 66 hex chars + null terminator.
+
+    // milliseconds since unix epoch:
+    int64_t last_read;
+
+    // expiration mode & time:
+    CONVO_EXPIRATION_MODE exp_mode;
+    int64_t exp_minutes;
+} convo_one_to_one;
+
+typedef struct convo_open_group {
+    const char* base_url; // null-terminated, always lower-case
+    const char* room; // null-terminated, always lower-case
```

Comment:
Is the `room` always lower case? (currently iOS forces the `base_url` to lowercase but not the `room`)

Reply:
Ah I saw the comment in `conversations.hpp` - I'll resolve this comment

Reply:
I don't think this is our definitive list of which open groups you are in; that will likely live in a "groups" config message (and would be where the appropriate case is stored).  But it is case-insensitive on the SOGS side, so it seemed better to lower-case it here on input so that regardless of how a local client manages it, we'll have a consistent value here.

Reply:
I guess I figured adding a separate "groups" config would be redundant since we have this (I'm not sure what else we'd store in there 😕)

Reply:
I could see us storing some metadata in there such as the time that you joined, maybe a local alias for the group, and so on.

Reply:
Yea ok, I guess my worry is that any functionality we add for open group conversations we'd also want for non-open-group conversations as well and just end up with `CONVERSATIONS` and `CONVERSATIONS_2` config types

Reply:
My idea is that we have this one for tracking highly volatile conversation details (like last read timestamp), which we keep as small as possible, then put the details that don't change often into a "groups" config that contains the open group, closed group, legacy closed group fixed data.  (This is similar to what we have now with Contacts holding contact slow-changing data, and the last read timestamp going in here).

Reply:
Mmk, did you want to shift the disappearing messages setting into that type instead of having it in here then? (It might also make sense to call that one `CONVERSATIONS` and rename this one to something like `CONVOS_LAST_READ` then)

Reply:
(or are you thinking that the disappearing messages setting might change frequently enough as well?)

Reply:
Although thinking about it some more, perhaps rather than storing URL + room + pubkey we should just store a (smaller) hash of that in here to keep size reduced.

So you'd go pull your list of open groups from the Groups config, then would look up H(url | room | pubkey) in here.

Reply:
> Mmk, did you want to shift the disappearing messages setting into that type instead of having it in here then?

Yeah, actually, that's probably worthwhile.  I do want to keep this one (which gets re-push frequently) as small as possible.

Reply:
No worries, I'll let the guys know that the conversation stuff is still in flux in case they happen to start working on it